x/crisis/keeper/keeper.go

```diff
@@ -84,6 +84,8 @@ func (k *Keeper) AssertInvariants(ctx sdk.Context) {
 	n := len(invarRoutes)
 	for i, ir := range invarRoutes {
 		logger.Info("asserting crisis invariants", "inv", fmt.Sprint(i+1, "/", n), "name", ir.FullRoute())
+
+		ctx, _ := ctx.CacheContext()
```

Comment:
Add `CacheContext()` inside the loop, otherwise, an invariant may affect the other's precondition.